libbeat/outputs/elasticsearch/client.go

```diff
@@ -718,15 +718,16 @@ func (conn *Connection) execHTTPRequest(req *http.Request) (int, []byte, error)
 	defer closing(resp.Body)
 
 	status := resp.StatusCode
+	var retErr error
 	if status >= 300 {
-		return status, nil, fmt.Errorf("%v", resp.Status)
+		retErr = fmt.Errorf("%v", resp.Status)
 	}
 
 	obj, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		return status, nil, err
```

Comment:
Hmm, debatable I guess. I'm changing it to `retErr` because I think that's closer to the previous behavior. Thanks.